pkg/userop/gas.go

```diff
@@ -0,0 +1,187 @@
+package userop
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/shopspring/decimal"
+)
+
+func getPolygonGasPrices(chainId *big.Int) (*big.Int, *big.Int, error) {
+	var resp *http.Response
+	var err error
+
+	switch {
+	case chainId.Cmp(big.NewInt(137)) == 0:
+		resp, err = http.Get("https://gasstation.polygon.technology/v2")
+	case chainId.Cmp(big.NewInt(80002)) == 0:
+		resp, err = http.Get("https://gasstation.polygon.technology/amoy")
+	default:
+		return nil, nil, fmt.Errorf("unsupported chain ID: %v", chainId)
+	}
```

Comment:
There's a way to simplify integer comparisons by casting `chainId` to `uint64` with `chainId.Uint64()`

Reply:
(a non-blocking suggestion) It would be cool to add conditional retries, if the call fails due to network congestion or other network-related issues

Reply:
Do you suggest retrying no matter what the error or retry only when received some specific errors?

Reply:
I see. Unfortunately, Polygon gas station docs do not provide any error reference, and all the calls made were successful. Let's continue without retries for now, but we will return to this suggestion if we encounter some errors.

Reply:
68cb8e64bff038323cf947fc2e017df1c8dafce5